src/uuid.cr

```diff
@@ -0,0 +1,251 @@
+# Universally Unique IDentifier.
+# Supports [RFC4122](https://www.ietf.org/rfc/rfc4122.txt) UUIDs and custom
+struct UUID
+  enum Variant # variants with 16 bytes.
+    Unknown    # Unknown (ie. custom, your own).
+    NCS        # Reserved by the NCS for backward compatibility.
+    RFC4122    # Reserved for RFC4122 Specification (default).
+    Microsoft  # Reserved by Microsoft for backward compatibility.
+    Future     # Reserved for future expansion.
+  end
+
+  enum Version  # RFC4122 UUID versions.
+    Unknown = 0 # Unknown version.
+    V1      = 1 # date-time and MAC address.
+    V2      = 2 # DCE security.
+    V3      = 3 # MD5 hash and namespace.
+    V4      = 4 # random.
+    V5      = 5 # SHA1 hash and namespace.
+  end
+
+  # Internal representation.
+  @bytes : StaticArray(UInt8, 16)
+
+  # Generates RFC 4122 v4 UUID.
+  def initialize
+    @bytes = uninitialized UInt8[16]
+    @bytes.to_unsafe.copy_from Random::Secure.random_bytes(16).pointer(16), 16
```

Comment:
Let's make it simpler, safer (no pointers), and faster (avoids a HEAP allocation):

```crystal
@bytes = uninitialized UInt8[16]
Random::Secure.random_bytes(@bytes.to_slice)
```